### DISCUSSION TITLE
MavenReportException: Error while generating Javadoc: Unable to write 'options' temporary file for command execution: Input length = 1

### DISCUSSION THREAD

**Discussion by QuantityIntel** [jplag/JPlag#399] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/46237445/160391075-df3763ad-4ed6-4e47-b330-6ad9a8a5c185.png)

I have the error above.

`$ java -version
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment Temurin-17.0.2+8 (build 17.0.2+8)
OpenJDK 64-Bit Server VM Temurin-17.0.2+8 (build 17.0.2+8, mixed mode, sharing)`

`
$ mvn -version
Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: C:\dev\apache-maven-3.8.5
Java version: 17.0.2, vendor: Eclipse Adoptium, runtime: C:\Program Files\Eclipse Adoptium\jdk-17.0.2.8-hotspot
Default locale: ko_KR, platform encoding: MS949
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"
`

What should I do?

**Comment by tsaglam** [comment]
What version of JPlag are you trying to build? Which mvn command did you run? We are not on Java 17 yet, so I have not tried building JPlag with Java 17 yet.

**Comment by QuantityIntel** [comment]
> What version of JPlag are you trying to build? Which mvn command did you run? We are not on Java 17 yet, so I have not tried building JPlag with Java 17 yet.

I'm trying to build v3.0.0.
I ran the command `mvn clean package`
Which version of Java would be compatible?

**Comment by tsaglam** [comment]
I have tested building v3.0.0 with Java 17 and works fine for me on my system. 
Maybe try `mvn clean package -U`. If it does not work please provide:

- the full CLI output of `mvn clean package`
- the output of `mvn -version`

**Comment by QuantityIntel** [comment]
> I have tested building v3.0.0 with Java 17 and works fine for me on my system. Maybe try `mvn clean package -U`. If it does not work please provide:
> 
> * the full CLI output of `mvn clean package`

<details>
  <summary><b>Click to expand!</b></summary>

`$ mvn clean package

[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for de.jplag:frontend-utils:jar:3.1.0-SNAPSHOT      
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for de.jplag:chars:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for de.jplag:cpp:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:csharp-1.2:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:java:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:python-3:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:scheme:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:text:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:jplag:jar:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 311, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 299, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ de.jplag:aggregator:3.1.0-SNAPSHOT, C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\pom.xml, line 287, column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective model for de.jplag:aggregator:pom:3.1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-gpg-plugin is missing. @ line 311, column 21        
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-source-plugin is missing. @ line 299, column 21     
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-javadoc-plugin is missing. @ line 287, column 21
[WARNING]
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING]
[INFO] Inspecting build with total of 10 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 10 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] JPlag Plagiarism Detector                                          [pom]
[INFO] frontend-utils                                                     [jar]
[INFO] chars                                                              [jar]
[INFO] cpp                                                                [jar]
[INFO] csharp-1.2                                                         [jar]
[INFO] java                                                               [jar]
[INFO] python-3                                                           [jar]
[INFO] scheme                                                             [jar]
[INFO] text                                                               [jar]
[INFO] jplag                                                              [jar]
[INFO] 
[INFO] ------------------------< de.jplag:aggregator >-------------------------
[INFO] Building JPlag Plagiarism Detector 3.1.0-SNAPSHOT                 [1/10]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ aggregator ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.3.0:add-source (add-source) @ aggregator ---
[INFO] Source directory: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\jplag\target\generated-sources\javacc 
added.
[INFO] Source directory: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\jplag\target\generated-sources\antlr added.
[INFO]
[INFO] --- maven-javadoc-plugin:3.3.2:jar (attach-javadocs) @ aggregator ---
[INFO] Not executing Javadoc as the project is not a Java classpath-capable package
[INFO]
[INFO] >>> maven-source-plugin:3.2.1:jar (attach-sources) > generate-sources @ aggregator >>>
[INFO]
[INFO] --- build-helper-maven-plugin:3.3.0:add-source (add-source) @ aggregator ---
[INFO] Source directory: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\jplag\target\generated-sources\javacc 
added.
[INFO] Source directory: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\jplag\target\generated-sources\antlr added.
[INFO]
[INFO] <<< maven-source-plugin:3.2.1:jar (attach-sources) < generate-sources @ aggregator <<<
[INFO]
[INFO]
[INFO] --- maven-source-plugin:3.2.1:jar (attach-sources) @ aggregator ---
[INFO] 
[INFO] ----------------------< de.jplag:frontend-utils >-----------------------
[INFO] Building frontend-utils 3.1.0-SNAPSHOT                            [2/10]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ frontend-utils ---
[INFO] Deleting C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\target
[INFO] 
[INFO] --- build-helper-maven-plugin:3.3.0:add-source (add-source) @ frontend-utils ---
[INFO] Source directory: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\target\generated-sources\javacc added.
[INFO] Source directory: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\target\generated-sources\antlr added.
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ frontend-utils ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\src\main\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ frontend-utils ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 7 source files to C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\target\classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ frontend-utils ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\src\test\resources
[INFO]
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ frontend-utils ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ frontend-utils ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:3.2.2:jar (default-jar) @ frontend-utils ---
[INFO] Building jar: C:\Users\kjaso\Dropbox\Shared Folder\Projects\plagiarism_checker\JPlag\jplag.frontend-utils\target\frontend-utils-3.1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-javadoc-plugin:3.3.2:jar (attach-javadocs) @ frontend-utils ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for JPlag Plagiarism Detector 3.1.0-SNAPSHOT:
[INFO]
[INFO] JPlag Plagiarism Detector .......................... SUCCESS [  1.329 s]
[INFO] frontend-utils ..................................... FAILURE [  2.109 s]
[INFO] chars .............................................. SKIPPED
[INFO] cpp ................................................ SKIPPED
[INFO] csharp-1.2 ......................................... SKIPPED
[INFO] java ............................................... SKIPPED
[INFO] python-3 ........................................... SKIPPED
[INFO] scheme ............................................. SKIPPED
[INFO] text ............................................... SKIPPED
[INFO] jplag .............................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.510 s
[INFO] Finished at: 2022-03-30T15:06:14+09:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.3.2:jar (attach-javadocs) on project frontend-utils: MavenReportException: Error while generating Javadoc: Unable to write 'options' temporary file for command execution: Input 
length = 1 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :frontend-utils`

</details>

> * the output of `mvn -version`

`$ mvn -version

Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: C:\dev\apache-maven-3.8.5
Java version: 17.0.2, vendor: Eclipse Adoptium, runtime: C:\Program Files\Eclipse Adoptium\jdk-17.0.2.8-hotspot
Default locale: ko_KR, platform encoding: MS949
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"`

**Comment by dfuchss** [comment]
@Jiseong-Michael-Yang do you have docker? Then you could test whether the compilation works within docker on your machine.
I've just tested the compilation process within the following docker image: `maven:3.8.4-eclipse-temurin-17` (v.3.8.5 hasn't released for docker so far)

Could you please run maven package with debug option to get more insights.

**Comment by tsaglam** [comment, marked as answer]
A workaround is to disable the maven javadoc plugin plugin in the pom.xml in the main project directory.
